server.go

```diff
@@ -64,6 +64,7 @@ func ParseServerConfig(path string) (*ServerConfig, error) {
 	}
 
 	reader, err := os.Open(path)
+	defer reader.Close()
```

Comment:
It needs to be after error check, otherwise when error happens it will cause panic (because reader would be nil)